File: frontend/src/lib/stores/robot.ts
import { writable } from 'svelte/store';

export interface RobotStatus {
  connected?: boolean;
  status: string;
  battery: number;
  mode: string;
  current_task?: string | null;
  current_state?: string;
}

export const robotStore = writable<RobotStatus>({
  connected: true,
  status: '待命',
  battery: 85,
  mode: '手动',
  current_task: null,
  current_state: '待命'
});

File: backend/app/models/robot.py
# 模型文件
# app/models/robot.py
# 定义机器狗相关的数据结构，包括：

# 机器狗的状态（RobotStatus）

# 控制指令（Command）

from pydantic import BaseModel
from typing import Optional

class RobotStatus(BaseModel):
    connected: bool
    battery: int
    current_state: str
    current_task: Optional[str] = None
    mode: str = "手动"

class Command(BaseModel):
    cmd: str
    speed: Optional[float] = 0.3
    duration: Optional[float] = 2.0